sdmetrics/reports/multi_table/_properties/base.py

```diff
@@ -45,6 +45,14 @@ def _get_num_iterations(self, metadata):
                 iterations += (len(parent_columns) * len(child_columns))
             return iterations
 
+    @staticmethod
+    def _extract_tuple(data, relation):
+        parent_data = data[relation['parent_table_name']]
+        child_data = data[relation['child_table_name']]
```

Comment:
should we error if the parent or child table is not in the data?

Reply:
I feel that this is the report's job to check that the relationships are correct.
We have this method:
https://github.com/sdv-dev/SDMetrics/blob/5f7488a35fd1ebd71e12d8fcd153e54815eb5839/sdmetrics/reports/multi_table/base_multi_table_report.py#L17
That does something a bit different but should error if the parent or child table is not in the data I think.
so when we call `_extract_tuple()` this has already been checked.

Reply:
BTW I think this method (`_validate_relationships`) is a bit old, maybe we want to update it?